h/views/account_signup.py

```diff
@@ -76,15 +81,172 @@ def validation_failure(self):
 
     @property
     def js_config(self) -> dict[str, Any]:
-        return {"csrfToken": get_csrf_token(self.request)}
+        feature_service = self.request.find_service(name="feature")
+
+        return {
+            "csrfToken": get_csrf_token(self.request),
+            "features": {
+                "log_in_with_orcid": feature_service.enabled(
+                    "log_in_with_orcid", user=None
+                )
+            },
+        }
+
+
+class IDInfoJWTDecodeError(Exception):
+    """Decoding the `idinfo` JWT query param failed."""
+
+
+@dataclass
+class IDInfo:
+    """Information about the user's account with a third-party provider.
+
+    When signing up to Hypothesis with a third-party provider ("Sign up with
+    Google" etc) the IDInfo class represents the payload of a signed JWT that
+    we use to pass information about the user's account with the third-party
+    from one route of our app to another when redirecting the browser (we
+    include the JWT in a query param).
+
+    The standard JWT `sub` (subject) claim is used to contain the user's unique
+    ID from the third-party provider. The JWT spec says that `sub` must be
+    locally unique within the context of the issuer (the `iss` claim). We
+    achieve this by using a different `iss` for each provider (Google,
+    Facebook, etc).
+
+    https://www.rfc-editor.org/rfc/rfc7519#section-4.1.2
+
+    """
+
+    sub: str
+
+
+@view_defaults(
+    route_name="signup.orcid",
+    is_authenticated=False,
+    request_param="idinfo",
+    renderer="h:templates/accounts/signup.html.jinja2",
+)
+class ORCIDSignupViews:
```

Comment:
The views for the new "signup with ORCID" interstitial signup page that we redirect you to after you authorize us to access your ORCID record. This page knows your ORCID iD and displays it at the top of the page, and asks you to choose a username and submit a form to sign up.

Nothing about this page/these views is specific to OIDC: it's just a signup page that happens to require an idinfo token in a query param (these idinfo JWTs are a custom thing, nothing to do with OIDC). Hence this code lives in `views/account_signup.py` alongside the normal signup views, not in `views/oidc.py`.